execution/executor-benchmark/src/account_generator.rs

```diff
@@ -78,42 +93,89 @@ impl AccountCache {
         let other = self.accounts.split_off(index);
-        (self.accounts.into(), other.into())
+        let accounts: Vec<LocalAccount> = self
+            .accounts
+            .into_iter()
+            .map(|a| Arc::try_unwrap(a).unwrap().into_inner())
+            .collect();
+        let other: Vec<LocalAccount> = other
+            .into_iter()
+            .map(|a| Arc::try_unwrap(a).unwrap().into_inner())
+            .collect();
+        (accounts, other)
     }
 
     pub fn len(&self) -> usize {
         self.accounts.len()
     }
 
-    pub fn accounts(&self) -> &VecDeque<LocalAccount> {
+    pub fn accounts(&self) -> &VecDeque<Arc<RwLock<LocalAccount>>> {
         &self.accounts
     }
 
     pub fn grow(&mut self, n: usize) {
-        let accounts: Vec<_> = (0..n).map(|_| self.generator.generate()).collect();
+        let accounts: Vec<_> = (0..n)
+            .map(|_| Arc::new(RwLock::new(self.generator.generate())))
+            .collect();
         self.accounts.extend(accounts);
     }
+}
 
-    pub fn get_random(&mut self) -> &mut LocalAccount {
-        let indices = rand::seq::index::sample(&mut self.rng, self.accounts.len(), 1);
-        let index = indices.index(0);
-
-        &mut self.accounts[index]
-    }
-
-    pub fn get_random_transfer_batch(
+impl RandomAccountGenerator for &mut AccountCache {
+    fn get_random_transfer_batch(
         &mut self,
         batch_size: usize,
-    ) -> (&mut LocalAccount, Vec<AccountAddress>) {
+    ) -> (Arc<RwLock<LocalAccount>>, Vec<AccountAddress>) {
         let indices = rand::seq::index::sample(&mut self.rng, self.accounts.len(), batch_size + 1);
         let sender_idx = indices.index(0);
         let receivers = indices
             .iter()
             .skip(1)
-            .map(|i| self.accounts[i].address())
+            .map(|i| self.accounts[i].read().address())
             .collect();
-        let sender = &mut self.accounts[sender_idx];
+        let sender = self.accounts[sender_idx].clone();
 
         (sender, receivers)
     }
+
+    fn get_random(&mut self) -> Arc<RwLock<LocalAccount>> {
+        let indices = rand::seq::index::sample(&mut self.rng, self.accounts.len(), 1);
+        let index = indices.index(0);
+
+        self.accounts[index].clone()
+    }
+}
+
+pub struct NoConflictsAccountCache {
+    pub accounts: VecDeque<Arc<RwLock<LocalAccount>>>,
+}
+
+impl NoConflictsAccountCache {
+    const SEED: Seed = [1; 32];
+
+    pub fn new(account_cache: &AccountCache) -> Self {
+        let mut rng = StdRng::from_seed(Self::SEED);
+        let mut accounts = account_cache.accounts.clone();
+        accounts.make_contiguous().shuffle(&mut rng);
+        Self { accounts }
+    }
+}
+
+impl RandomAccountGenerator for NoConflictsAccountCache {
+    fn get_random(&mut self) -> Arc<RwLock<LocalAccount>> {
+        // Since the accounts are already shuffled, we can just pop the first one.
+        self.accounts.pop_front().unwrap()
```

Comment:
That's a very good observation @gelash - I actually thought about it but I left it as `Arc` as I will have a follow up to parallelize the txn generation part - I am seeing it not able to generate more than 25k TPS on single threaded executor.  That change will require this to change to thread-safe semantic anways, so I prefer to leave it as is to avoid duplicate work. LMK your thoughts.

Reply:
I see, that's a good consideration. I do wonder how the parallel generator would look like, or whether some sort of freezing / cloning / other tricks could still allow different interfaces, but this question can be revisited at that point. I think it's fine to keep it as is then.

Reply:
Why not use the transaction-generator mechanisms here?

It has parallel/saturating account creation, it has pool of accounts, and allows for CoinTransferUnique?

Reply:
If we didn't need it for @grao1991's native executor special db plans, I would completely remove non-transaction-generator-lib way of generating transactions

Reply:
@igor-aptos - Good point, can you clarify if `CoinTransferUnique` allows unique sender and receiver in a block? Also, I don't see any way to reset the account pool for every block? Does this mean we have to generate very high number of unique accounts, which might not work for longer running benchmark.

Reply:
I would suggest iterating on transaction-generator-lib to support new needs, as a preferred method :)

for the sender to be unique, just use transactions_per_sender = 1.

For what you need, I would change P2PTransactionGenerator to have additional flag - "burn_and_recycle_targets", to in addition of:
    all_addresses: Arc<RwLock<Vec<AccountAddress>>>,
have:
    burnt_addresses: Arc<RwLock<Vec<AccountAddress>>>,
where the used addresses will go, and once all_addresses doesn't have enough, it moves all burnt_addresses back

and so that is even more flexible - as long as the pool is bigger than a 2 blocks, you have that guarantee you need

Reply:
@igor-aptos @sitalkedia 
how about a new trait `BlockGenerator` with a function like: `generate_block(&mut self, block_size: usize, transactions_per_sender: usize, no_conflicting_txns: bool)`.

Reply:
@zjma - A new `BlockGenerator` trait sounds good - one thing to note is that this trait will only be used for the executor benchmark. Other use cases of transaction generator lib (like Forge) don't really care/know about the block boundary. All we need to do is keep track of `burnt_addresses` and at the beginning of the block, we recycle all the `burnt_addresses`.